identity-issuance/src/main/java/com/android/identity/issuance/wallet/ApplicationSupportState.kt

```diff
@@ -87,14 +92,86 @@ class ApplicationSupportState(
         return createJwtClientAssertion(env, attestation.publicKey, targetIssuanceUrl)
     }
 
+    @FlowMethod
+    fun createJwtKeyAttestation(
+        env: FlowEnvironment,
+        keyAttestations: List<KeyAttestation>,
+        nonce: String
+    ): String {
+        val settings = WalletServerSettings(env.getInterface(Configuration::class)!!)
+
+        val keyList = keyAttestations.map { attestation ->
+            // TODO: ensure that keys come from the same device and extract data for key_type
+            // and user_authentication values
+            validateKeyAttestation(
+                attestation.certChain!!,
+                nonce,
+                settings.androidRequireGmsAttestation,
+                settings.androidRequireVerifiedBootGreen,
+                settings.androidRequireAppSignatureCertificateDigests
+            )
+            attestation.publicKey.toJson(null)
+        }
+
+        val attestationData = env.cache(AttestationData::class) { configuration, resources ->
+            // The key that we use here is unique for a particular Wallet ecosystem.
+            // Use client attestation key as default for development (default is NOT suitable
+            // for production, as private key CANNOT be in the source repository).
+            val certificateName = configuration.getValue("openid4vci.key-attestation.certificate")
+                ?: "attestation/certificate.pem"
+            val certificate = X509Cert.fromPem(resources.getStringResource(certificateName)!!)
+            val privateKeyName = configuration.getValue("openid4vci.key-attestation.privateKey")
+                ?: "attestation/private_key.pem"
+            val privateKey = EcPrivateKey.fromPem(
+                resources.getStringResource(privateKeyName)!!,
+                certificate.ecPublicKey
+            )
+            val issuer = configuration.getValue("openid4vci.key-attestation.issuer")
+                ?: configuration.getValue("base_url")
+                ?: "https://github.com/openwallet-foundation-labs/identity-credential"
+            AttestationData(certificate, privateKey, issuer)
+        }
+        val publicKey = attestationData.certificate.ecPublicKey
+        val privateKey = attestationData.privateKey
+        val alg = publicKey.curve.defaultSigningAlgorithm.jwseAlgorithmIdentifier
+        val head = buildJsonObject {
+            put("typ", JsonPrimitive("keyattestation+jwt"))
+            put("alg", JsonPrimitive(alg))
+            put("jwk", publicKey.toJson(null))  // TODO: use x5c instead here?
+        }.toString().toByteArray().toBase64Url()
+
+        val now = Clock.System.now()
+        val notBefore = now - 1.seconds
+        val expiration = now + 5.minutes
```

Comment:
Are there any scenarios where there would be user interaction to make use of this object? If so, 5 minutes may not be enough (given clock skew and distracted people). ...If not, 5 minutes seems fine.

Reply:
Will add a comment. In funke example they set it for 30 seconds which I think is pushing it (as it needs to cross the network twice).